optaplanner-core/src/test/java/org/optaplanner/core/config/util/ConfigUtilsTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.config.util;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+import static org.optaplanner.core.config.util.ConfigUtils.*;
+
+public class ConfigUtilsTest {
+
+    @Test
+    public void testMergeProperty() {
+        Integer a = null;
+        Integer b = null;
+        assertEquals(null, mergeProperty(a, b));
+        a = Integer.valueOf(1);
+        assertEquals(null, mergeProperty(a, b));
+        b = Integer.valueOf(10);
+        assertEquals(null, mergeProperty(a, b));
+        b = Integer.valueOf(1);
+        assertEquals(Integer.valueOf(1), mergeProperty(a, b));
+        a = null;
+        assertEquals(null, mergeProperty(a, b));
+    }
+
+    @Test
+    public void testMeldProperty() {
+        Integer a = null;
+        Integer b = null;
+        assertEquals(null, meldProperty(a, b));
+        a = Integer.valueOf(1);
+        assertEquals(Integer.valueOf(1), meldProperty(a, b));
+        b = Integer.valueOf(10);
+        assertEquals(mergeProperty(Integer.valueOf(1), Integer.valueOf(10)), meldProperty(a, b));
+        a = null;
+        assertEquals(Integer.valueOf(10), meldProperty(a, b));
+    }
+
+    private static void checkCeilDivision(int dividend, int divisor) {
+        long result = Math.round(Math.ceil(dividend / (double) divisor));
```

Comment:
Although I disagree (mainly because I wrote the tests first and then checked the implementation against the tests), I see your point of view on this. The coverage is exactly the same as I had with this sort of test implementation (apart from the test logic, ofc).

This is your call, do I change it back or keep what we have?